.github/workflows/ci.yml

```diff
@@ -100,11 +100,9 @@ jobs:
         run: find . -iname "*.[hc]" -exec uncrustify --check -c tools/uncrustify.cfg -l C {} +
       - name: Check For Trailing Whitespace
         run: |
-          set +e
-          grep --exclude="README.md" -rnI -e "[[:blank:]]$" .
-          if [ "$?" = "0" ]; then
-            echo "Files have trailing whitespace."
-            exit 1
-          else
-            exit 0
-          fi
+          set +me
+          shopt -s lastpipe
```

Comment:
Are there gaps in the older logic? If yes, we would need to update all the repos using that logic..

Reply:
Yes, the old logic descends into the .git directory where it was finding trailing spaces.
```
$ grep --exclude="README.md" -rnI -e "[[:blank:]]$" . | cat -tve
./.git/packed-refs:1:# pack-refs with: peeled fully-peeled sorted $
```

Reply:
Ohh, we will need to update the logic elsewhere